src/server/educandu-server.js

```diff
@@ -58,34 +75,40 @@ class EducanduServer {
     controllers.filter(c => c.registerErrorHandler).forEach(c => c.registerErrorHandler(router));
   }
 
-  listen(cb) {
+  exitMaintenanceMode() {
+    this.app.locals[SYMBOL_MAINTENACE] = false;
+  }
+
+  listen({ maintenance = false }) {
+    this.app.locals[SYMBOL_MAINTENACE] = !!maintenance;
+    return new Promise((resolve, reject) => {
+      const { port } = this.serverConfig;
+      if (this.server) {
+        logger.info('Cannot start server, it is already listining');
+        resolve(port);
+      } else {
+        logger.info('Starting server');
+        this.server = this.app.listen(port, err => err ? reject(err) : resolve(port));
+      }
+    });
+  }
+
+  async close() {
     if (this.server) {
-      logger.info('Cannot start server, it is already listining');
-      Promise.resolve().then(() => cb(null, this.serverConfig.port));
+      logger.info('Closing server');
+      await util.promisify(this.server.close.bind(this.server))();
+      this.server = null;
+      logger.info('Server successfully closed');
     } else {
-      logger.info('Starting server');
-      this.server = this.app.listen(this.serverConfig.port, err => err ? cb(err) : cb(null, this.serverConfig.port));
+      logger.info('Cannot close server, it is not listening');
+      await Promise.resolve();
```

Comment:
I think this is not needed here.

Reply:
Historical code, I removed the whole `else` block, because also the log message is not really helpful.